File: backend/src/main/java/edu/ben/backend/model/Song.java
package edu.ben.backend.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@Data
@ToString
@EqualsAndHashCode
@AllArgsConstructor
public class Song {
    long id;
    String deezerUrl;
    String title;
    String artist;
    int averageRating;
    //album title
}

File: lib/models/song.dart
class Song {
  int id;
  String deezerUrl;
  String title;
  String artist;
  int rating;

  Song({
    required this.id,
    required this.deezerUrl,
    required this.title,
    required this.artist,
    required this.rating,
  });

  getNotes() {
    return "🎵" * rating;
  }
}
